### PR TITLE
 Adding 2 more arguments for add_portgroup_to_vswitch.py

### PR DESCRIPTION
**Pull request by samz-github**
1. Add argument to support whether SSL verified context or not
    2. Add argument to support regex match specific VMware esxi name

### CODE REVIEW

File: samples/add_portgroup_to_vswitch.py [hunk 1]
```diff
@@ -55,17 +57,39 @@ def get_args():
                         action='store',
                         help='Vlan ID')
 
+    parser.add_argument('-c', '--ssl_verified_context',
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: How about change the parameter to --skip_verificiation and default to False?

File: samples/add_portgroup_to_vswitch.py [hunk 2]
```diff
@@ -55,17 +57,39 @@ def get_args():
...
+                        default=False,
+                        action='store',
+                        help='Regex esxi name')
+
     args = parser.parse_args()
     return args
 
 
-def GetVMHosts(content):
+def GetVMHosts(content, regex_esxi):
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: Set default value to None for regex_esxi?

File: samples/add_portgroup_to_vswitch.py [hunk 3]
```diff
@@ -55,17 +57,39 @@ def get_args():
...
     obj = [host for host in host_view.view]
-    host_view.Destroy()
-    return obj
+    match_obj = []
+    if regex_esxi:
+        for esxi in obj:
+            if re.findall(r'%s.*' % regex_esxi, esxi.name):
+                match_obj.append(esxi)
+        match_obj_name = [match_esxi.name for match_esxi in match_obj]
+        print(match_obj_name)
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: Shall we print some message before the obj names?

File: samples/add_portgroup_to_vswitch.py [hunk 4]
```diff
@@ -90,14 +114,22 @@ def AddHostPortgroup(host, vswitchName, portgroupName, vlanId):
 
 def main():
     args = get_args()
-    serviceInstance = SmartConnect(host=args.host,
-                                   user=args.user,
-                                   pwd=args.password,
-                                   port=443)
+    if args.ssl_verified_context == 'True':
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: When skip_verification is true, you can use SmartConnectNoSSL.

File: samples/add_portgroup_to_vswitch.py [hunk 5]
```diff
@@ -55,17 +57,39 @@ def get_args():
                         action='store',
                         help='Vlan ID')
 
+    parser.add_argument('-c', '--ssl_verified_context',
+                        required=False,
+                        default='True',
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: You can use "action='store_true'" for boolean parameter.

File: samples/add_portgroup_to_vswitch.py [hunk 6]
```diff
@@ -55,17 +56,38 @@ def get_args():
...
+                        default=None,
+                        action='store',
+                        help='Regex esxi name')
+
     args = parser.parse_args()
     return args
 
 
-def GetVMHosts(content):
+def GetVMHosts(content, regex_esxi):
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: The method parameter should have None as the default value:
def GetVMHosts(content, regex_esxi=None):

File: samples/add_portgroup_to_vswitch.py [hunk 7]
```diff
@@ -8,11 +8,12 @@
 http://opensource.org/licenses/Apache-2.0
 """
 from __future__ import print_function
-from pyVim.connect import SmartConnect, Disconnect
+from pyVim.connect import SmartConnect, SmartConnectNoSSL, Disconnect
```
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: Could you please clean up the not used imports?

Comment (samz-github) [inline samples/add_portgroup_to_vswitch.py]: 1. Post output:
Matched ESXi hosts: ['esxi01.test.com']
2. Looks all imports are required.

File: samples/add_portgroup_to_vswitch.py (hunk 6 above)
Comment (tianhao64) [inline samples/add_portgroup_to_vswitch.py]: Can you fix this?

**Comment by tianhao64** [comment]
Ideally there should be one commit per PR to keep the history clean. Can you squash your commits? Thanks!

**Comment by samz-github** [comment]
Re-based commit messages, now it's clean now.

**Pull request merged**